remove interpolated points of dashboard's runtime bench and fix graph unit

s7tya opened the pull request:
I fixed the graph unit from seconds to nanoseconds and removed interpolated points from when runtime benchmarks hadn't been implemented yet.

Kobzol commented on site/frontend/src/pages/dashboard.ts:
```diff
@@ -92,7 +92,7 @@ function renderRuntime(element: string, data: [number], versions: [string]) {
     },
     yAxis: {
-      title: {text: "Seconds"},
+      title: {text: "Nanoseconds"},
```
I would actually prefer to just show a larger unit (milliseconds?), and then multiply the values, rather than showing these super big numbers.

Kobzol commented on site/src/request_handlers/dashboard.rs:
```diff
@@ -153,6 +155,8 @@ pub async fn handle_dashboard(ctxt: Arc<SiteCtxt>) -> ServerResult<dashboard::Re
                 ArtifactId::Commit(c) => format!("master: {}", &c.sha.to_string()[0..8]),
                 ArtifactId::Tag(aid) => aid.clone(),
             })
+            .collect::<Vec<_>>()
```
Correct me if I'm wrong, but this would also affect the compile time benchmarks, and we definitely don't want to do that, because that would essentially skip compile time versions for which we do not have runtime data.

I think it would be better to return e.g. something like `Vec<Option<f64>>` for the runtime data, and then let the frontend skip the compiler versions for which there is no data available.

Kobzol reviewed (commented):
Thank you for the PR, I left some comments. Are you able to test this locally, btw? (you can do that by manually modifying the results.db file to insert results that will be shown in the dashboard, or modify the querying code so that it also returns non-stable artifacts). Without testing, it will be quite challenging to get this right.

s7tya commented:
Thanks for the heads-up. I've been testing them locally, but only the data related to the runtime graphs were showing up, so I didn't catch my mistake. I just modified a few artifact names to "1.xx.x" and changed the type to "release." It only worked for the runtime data.

Kobzol commented:
Hmm, if the `versions` array contained some compiler versions, then your changes were probably fine, but the question is whether there was some compile benchmark data in your local DB. In any case, it would be great to simulate the situation where you have e.g. 5 compile-time artifacts, but only e.g. 3 runtime artifacts, to see if the missing runtime data is being ignored.

s7tya commented:
(I just updated the and the change may fix the problem, i'm still investigating why my local artifacts doesn't have compile benchmark)

Kobzol commented on site/frontend/src/pages/dashboard.ts:
```diff
@@ -80,6 +80,12 @@ function render(
+  const formattedData = data
+    .filter((data) => data != null)
+    .map((data) => data / 1_000_000);
+  const nullCount = data.length - formattedData.length;
```
This assumes that nulls be only at the beginning, but that should be fine in our case. It would be nice to add a comment to explain this invariant here though.

Kobzol commented on site/src/request_handlers/dashboard.rs:
```diff
@@ -137,13 +137,25 @@ pub async fn handle_dashboard(ctxt: Arc<SiteCtxt>) -> ServerResult<dashboard::Re
         .statistic_series(runtime_benchmark_query.clone(), aids.clone())
         .await?;
+
+    let mut ignore_runtime_benchmark = true;
```
This bool is needed if there is some interpolated data *after* the first runtime benchmark result?

As above, please add a comment here that explains why do we need this handling (because historically, we don't have runtime benchmark data for old artifacts).

Kobzol reviewed (commented):
Thank you! Left two comments that ask you to add two comments :laughing:

Kobzol reviewed (approved):
Thanks! Given that this is non-trivial to test locally, let's see how it will look like on the live site.

s7tya commented:
Thanks!
